### DISCUSSION TITLE
Including Plotly plots in a post

### DISCUSSION THREAD

**Discussion by AxelBogos** [alshedivat/al-folio#523] [Q&A] [answered]
Hi! I hoped to know if someone has found a more elegant solution to the issue I'm having.
## Background 
I wish to include a Plotly figure as an html file in a blogpost. I have added my plot in the `_includes` directory. In typical Jekyll/Markdown blogs, this can be done as `{% include dynamic_plot.html %}` in a blog post. The following issues occur here: 
## Issues
* Issue 1 - Jemoji
Including a figure in this way seems to raise a conflict with Jemoji (similar issue reported [here](https://talk.jekyllrb.com/t/insert-interactive-plotly-figure-in-jekyll-website/6481)). I have not found another solution than removing Jemoji; thats something that I can live with on that blog but it would be nice to keep.
* Issue 2 - Layout breaking
More critically, including the figure in this way breaks the layout of both `distill` or `post` styled posts. As an example: 
 
<img width="1348" alt="Screen Shot 2022-01-05 at 7 52 01 PM" src="https://user-images.githubusercontent.com/35436065/148313683-7da9ff71-843e-4bcf-a1f6-14805d8bd757.png">

I have tried surrounding it within a `<div>` as is done with images in the blog examples, to no avail. 
## Workaround
Embedding the plot with an `<iframe>` works, you can see a [dynamic demo here](https://axelbogos.github.io/blog/2021/covid_hosp_cases_by_vacc_status/), but it is unworkable with larger plots as the overhead to embed heavier plots is significant : ~45s for some plots vs ~10s for the "normal" way of including dynamic plots in jekyll. 

My repo is [here](https://github.com/AxelBogos/AxelBogos.github.io) if that is of use to anyone. 
I'm open to any suggestion or ideas of people having encountered a similar problem 🙏 !

**Comment by matthias-br** [comment, marked as answer]
I've found that Jekyll does not like it when there are underscores in file or folder names. For me, that resulted in plotly graphs not showing. Removing underscores fixed it. Perhaps it can solve your issues, too.
